XCalendar.Core/Extensions/StringsExtensions.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace XCalendar.Core.Extensions
+{
+    public static class StringsExtensions
+    {
+        public static string TruncateStringToMaxLength(this string value, object parameter)
+        {
+            int maxLength = 0;
+
+            bool parameterIsNotNullAndIsAInt = parameter != null && int.TryParse(parameter.ToString(), out maxLength);
+
+            if (parameterIsNotNullAndIsAInt)
+            {
+                return value.TruncateStringToMaxLength(maxLength);
+            }
+
+            return string.Empty;
+        }
+
+        public static string TruncateStringToMaxLength(this string value, int maxLength)
+        {
+            if (value == null)
+            {
+                return string.Empty;
+            }
+
+            if (maxLength <= 0)
+            {
+                return string.Empty;
+            }
+            else if (maxLength >= value.Length)
+            {
+                return value;
+            }
+            else
+            {
+                return value.Substring(0, maxLength);
+            }
+        }
+
+        public static string UppercaseFirst(this string text)
+        {
+            if (string.IsNullOrEmpty(text))
+            {
+                return string.Empty;
+            }
+
+            return char.ToUpper(text[0]) + text.Substring(1);
+        }
```

Comment:
The "ToUpper" method allows for a `CultureInfo` parameter. Since this isn't used, in cases where you localise the string first and then uppercase it, the culture may not be preserved. For example when it is used in `LocalizeDayOfWeekAndCharLimitConverter` where a specific culture is requested to be used.

Similar to .NET's "ToUpper" method, an overload should be added with an additional parameter for the culture.

Reply:
I refaactor and now i use culture.TextInfo.ToTitleCase(text) that is the best option for this case.